src/equations/acoustic_perturbation_2d.jl

```diff
@@ -127,12 +127,13 @@ A smooth initial condition used for convergence tests in combination with
 """
 function initial_condition_convergence_test(x, t,
                                             equations::AcousticPerturbationEquations2D)
-    c = 2.0
-    A = 0.2
-    L = 2.0
-    f = 2.0 / L
-    a = 1.0
-    omega = 2 * pi * f
+    RealT = eltype(x)
+    a = 1
+    c = 2
+    L = convert(RealT, 2) # since we divide by L below
+    f = 2 / L
```

Comment:
Should we really jump through the hoop of using `convert(RealT, 2)` here? Hypothetically, if this setup would use a constant `c = 1.5`, wouldn't we write `c = 1.5f0` here and then force everything to FP32 anyways?

Since we will be supporting only single precision with this revamp anyways, why not just write
```suggestion
    L = 2
    f = 2.0f0 / L
```
here and in similar locations elsewhere?

Reply:
Fine with me